lib/mpl_toolkits/axes_grid1/axes_grid.py

```diff
@@ -38,21 +43,30 @@ def colorbar(self, mappable, *, ticks=None, **kwargs):
             if ticks is None:
                 ticks = ticker.MaxNLocator(5)  # For backcompat.
             from .colorbar import Colorbar
+            cb = Colorbar(
+                self, mappable, orientation=orientation, ticks=ticks, **kwargs)
+            self._cbid = mappable.callbacksSM.connect(
+                'changed', cb.update_normal)
+            mappable.colorbar = cb
+            self._locator = cb.cbar_axis.get_major_locator()  # deprecated.
         else:
-            from matplotlib.colorbar import Colorbar
-        cb = Colorbar(
-            self, mappable, orientation=orientation, ticks=ticks, **kwargs)
-        self._config_axes()
+            cb = mpl.colorbar.colorbar_factory(
+                self, mappable, orientation=orientation, ticks=ticks, **kwargs)
+            self._cbid = mappable.colorbar_cid  # deprecated.
+            self._locator = cb.locator  # deprecated.
 
-        self.cbid = mappable.callbacksSM.connect('changed', cb.update_normal)
-        mappable.colorbar = cb
+        self._config_axes()
+        return cb
 
-        if mpl.rcParams["mpl_toolkits.legacy_colorbar"]:
-            self.locator = cb.cbar_axis.get_major_locator()
-        else:
-            self.locator = cb.locator
+    @cbook.deprecated("3.3", alternative="mappable.colorbar_cid")
+    @property
+    def cbid(self):
+        return self._cbid
 
-        return cb
+    @cbook.deprecated("3.3", alternative="colorbar.locator")
```

Comment:
What is `colorbar`? I don't see that attribute/property.

Reply:
It's the locator attribute of the colorbar object returned by the colorbar() method -- just like above mappable.colorbar_cid is the colorbar_cid attribute of the mappable object passed to colorbar().  I don't think for standard colorbars there's an easy way to go from the axes to the colorbar if you didn't explicitly get a hold on it, perhaps there should, but given that no one asked for one for standard colorbars, I don't think it's necessary for mpl_toolkits colorbars either.

Reply:
So it should be `.colorbar().locator` then? It's just not clear to me what to use if accessing `CbarAxesBase.locator` and getting a deprecation warning saying to use `colorbar.locator`, when I don't have a `colorbar`.

Reply:
sure, changed it to .colorbar().locator.